### PR TITLE
Added EmojiUtils

### PR DESCRIPTION
**Pull request by aballano**
This class contains 2 methods for now:
* isOnlyEmojis: Returns true wether the given text only contains emojis (whitespaces are ignored)
* emojisCount: Returns the number of emojis for the given text.

### CODE REVIEW

File: emoji/src/main/java/com/vanniktech/emoji/EmojiHandler.java [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 import android.content.Context;
 import android.text.Spannable;
+
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiHandler.java]: nit: remove these

File: emoji/src/main/java/com/vanniktech/emoji/EmojiHandler.java [hunk 2]
```diff
@@ -2,7 +2,9 @@
...
 import com.vanniktech.emoji.EmojiManager.EmojiRange;
+
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiHandler.java]: nit: remove these

File: emoji/src/main/java/com/vanniktech/emoji/EmojiManager.java [hunk 3]
```diff
@@ -2,8 +2,10 @@
 
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
+
 import com.vanniktech.emoji.emoji.Emoji;
 import com.vanniktech.emoji.emoji.EmojiCategory;
+
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiManager.java]: nit: remove these

File: emoji/src/main/java/com/vanniktech/emoji/EmojiManager.java [hunk 4]
```diff
@@ -2,8 +2,10 @@
 
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
+
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiManager.java]: nit: remove these

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package com.vanniktech.emoji;
+
+import android.support.annotation.Nullable;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class EmojiUtils {
+
+  private static final Pattern SPACE_REMOVAL = Pattern.compile(" ");
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: what about tabs?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
+package com.vanniktech.emoji;
+
+import android.support.annotation.Nullable;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class EmojiUtils {
+
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: nit: remove

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package com.vanniktech.emoji;
+
+import android.support.annotation.Nullable;
+
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: nit: remove

File: emoji/src/main/java/com/vanniktech/emoji/Utils.java [hunk 8]
```diff
@@ -106,4 +106,8 @@ static void fixPopupLocation(@NonNull final PopupWindow popupWindow, @NonNull fi
   private Utils() {
     throw new AssertionError("No instances.");
   }
+
+  static boolean isEmpty(@Nullable String text) {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/Utils.java]: `TextUtils.isEmpty`?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class EmojiUtils {
+
+  private static final Pattern SPACE_REMOVAL = Pattern.compile(" ");
+
+  static boolean isOnlyEmojis(@Nullable String text) {
+    if (Utils.isEmpty(text)) return false;
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: can you add braces?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+import android.support.annotation.Nullable;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class EmojiUtils {
+
+  private static final Pattern SPACE_REMOVAL = Pattern.compile(" ");
+
+  static boolean isOnlyEmojis(@Nullable String text) {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: should be public if you want to access it

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+    final String inputWithoutSpaces = SPACE_REMOVAL.matcher(text.trim()).replaceAll(Matcher.quoteReplacement(""));
+
+    return EmojiManager.getInstance()
+          .getEmojiRepetitivePattern()
+          .matcher(inputWithoutSpaces)
+          .matches();
+  }
+
+  static int emojisCount(@Nullable String text) {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: should be public if you want to access it

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.regex.Pattern;
+
+public final class EmojiUtils {
+
+  private static final Pattern SPACE_REMOVAL = Pattern.compile(" ");
+
+  static boolean isOnlyEmojis(@Nullable String text) {
+    if (Utils.isEmpty(text)) return false;
+
+    final String inputWithoutSpaces = SPACE_REMOVAL.matcher(text.trim()).replaceAll(Matcher.quoteReplacement(""));
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: Do we need to trim if we anyways remove all of the spaces?

File: emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+import com.vanniktech.emoji.emoji.Emoji;
+import com.vanniktech.emoji.emoji.EmojiCategory;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class EmojiUtilsTest {
+
```
Comment (vanniktech) [inline emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java]: nit: remove

File: emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java [hunk 14]
```diff
@@ -0,0 +1,165 @@
+package com.vanniktech.emoji;
+
+import android.support.annotation.NonNull;
+
+import com.pushtorefresh.private_constructor_checker.PrivateConstructorChecker;
+import com.vanniktech.emoji.emoji.Emoji;
+import com.vanniktech.emoji.emoji.EmojiCategory;
+
```
Comment (vanniktech) [inline emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java]: nit: remove

File: emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java [hunk 15]
```diff
@@ -0,0 +1,165 @@
+package com.vanniktech.emoji;
+
+import android.support.annotation.NonNull;
+
```
Comment (vanniktech) [inline emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java]: nit: remove

File: emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class EmojiUtilsTest {
+
+  private static final String EMOJI_1 = "\u1234";
+  private static final String EMOJI_2 = "\u4321";
+
+  @Before public void setUp() {
+    EmojiManager.install(new EmojiProvider() {
+      @NonNull
```
Comment (vanniktech) [inline emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java]: nit: same line

File: emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+
+  @Test public void constructorShouldBePrivate() {
+    PrivateConstructorChecker.forClass(EmojiUtils.class)
+          .expectedTypeOfException(AssertionError.class)
+          .expectedExceptionMessage("No instances.")
+          .check();
+  }
+
+  @Test public void isOnlyEmojis_empty() {
+    boolean onlyEmojis = EmojiUtils.isOnlyEmojis("");
```
Comment (vanniktech) [inline emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java]: just inline all of these variables please

File: emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  @Test public void constructorShouldBePrivate() {
+    PrivateConstructorChecker.forClass(EmojiUtils.class)
+          .expectedTypeOfException(AssertionError.class)
+          .expectedExceptionMessage("No instances.")
+          .check();
+  }
+
+  @Test public void isOnlyEmojis_empty() {
```
Comment (vanniktech) [inline emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java]: also remove underscores from method names

File: emoji/src/main/java/com/vanniktech/emoji/Utils.java (hunk 8 above)
Comment (aballano) [inline emoji/src/main/java/com/vanniktech/emoji/Utils.java]: I can use that indeed, just decided not to so I don't have to use Robolectric for the test. As you prefer :)

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java (hunk 5 above)
Comment (aballano) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: Also added line terminations to get rid of the `trim()`

File: emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java [hunk 19]
```diff
@@ -0,0 +1,32 @@
+package com.vanniktech.emoji;
+
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class EmojiUtils {
```
Comment (rubengees) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: I think the `EmojiManager` should provide this functionality and not another class.

@vanniktech Maybe we should make all of the `EmojiManager` public now as with this included changes it is quite usefull for other things apart from only managing `TextViews` and `EditTexts`.

Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiUtils.java]: Let's leave it here for now (in EmojiUtils) I anyways thought of restructuring it a bit before the 0.5.0 release.

**Comment by aballano** [comment]
@vanniktech 

`[ant:checkstyle] [ERROR] /home/travis/build/vanniktech/Emoji/emoji/src/test/java/com/vanniktech/emoji/EmojiUtilsTest.java:70:50: The String "hello" appears 8 times in the file. [MultipleStringLiterals]`

Really? :D

**Comment by aballano** [comment]
@vanniktech I added a suppression for that MultipleStringLiterals in tests, let me know if that's not ok.

**Pull request merged**